warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,227 @@
+     *
+     * @param config
+     *            the Hadoop Configuration containing properties like <code>typeName.whindex.rules.X.value_field</code>
+     * @throws RuntimeException
+     *             if any property is malformed or missing required values
+     */
+    @Override
+    public void setup(Configuration config) {
+        try {
+            String prefix = type.typeName() + "." + WHINDEX_RULES + ".";
+            Map<String,String> properties = config.getPropsWithPrefix(prefix);
+            Map<String,WhindexConfig.Builder> builders = new HashMap<>();
+
+            for (Map.Entry<String,String> entry : properties.entrySet()) {
+                String key = entry.getKey();
+                String val = entry.getValue();
+                String[] parts = key.split("\\.");
+                String group = parts[0];
+                String prop = parts[1];
+
+                if (val.isEmpty()) {
+                    throw new IllegalArgumentException("Empty value for Whindex property: " + key);
+                }
+
+                WhindexConfig.Builder builder = builders.computeIfAbsent(group, g -> WhindexConfig.builder());
+
+                switch (prop) {
+                    case VALUE_FIELD:
+                        builder.withValueField(val);
+                        break;
+
+                    case SRC_FIELD:
+                        builder.withSourceField(val);
+                        break;
+
+                    case DELETE_SRC_FIELD:
+                        builder.withOverloaded(Boolean.parseBoolean(val));
+                        break;
+
+                    case DST_FIELD:
+                        builder.withDestField(val);
+                        destinationFields.add(val);
+                        break;
+
+                    case VALUES:
+                        List<String> vals = List.of(val.split(","));
+                        builder.withValues(vals);
+                        break;
+
+                    default:
+                        throw new IllegalArgumentException("Unexpected Whindex property: " + key);
+                }
+            }
+
+            // Build all configs and populate the multimap
+            for (WhindexConfig cfg : builders.values().stream().map(WhindexConfig.Builder::build).collect(Collectors.toList())) {
+                valueFieldsToWhindexConfigs.put(cfg.getValueField(), cfg);
+            }
+
+            log.debug("Loaded Whindex configs: " + valueFieldsToWhindexConfigs.entries());
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to setup WhindexFieldIngestHelper", e);
+        }
+    }
+
+    /**
+     * Applies all configured Whindex rules to the given event map. For each config whose <code>valueField</code> has a triggering value, copies all entries
+     * from the <code>sourceField</code> into the new <code>destField</code>, optionally deleting the source.
```

Comment:
Optionally deleting the source when? Let's be a bit more specific here about what triggers the deletion of the source field (i.e. when the source is marked as overloaded in the config, and only when a whindex entry was created.)